packages/client/src/Signer.ts

```diff
@@ -174,8 +174,12 @@ export class KeyringSigner extends BaseSigner {
 
     private keyring: Keyring;
 
-    async signToHex(signerId: string, message: string): Promise<TypedSignature> {
-        const keypair = this.keyring.getPair(signerId);
+    async signToHex(signerId: ValidAccountId, message: string): Promise<TypedSignature> {
+        if(signerId.type !== "Polkadot") {
+            throw new Error(`Unsupported account type ${signerId.type}`);
+        }
```

Comment:
Correct me if I am wrong but I think this will prevent user to sign with Metamask.

Reply:
Yes, because that part still needs to be written.

Reply:
But it is already used when authenticating an Ethereum address with MetaMask in the certificate/restricted delivery. As such, this commit introduces a regression.

Reply:
Indeed, I misread that. I remove that part.